src/transports/winhttp.c

```diff
@@ -40,6 +40,19 @@
 #define WINHTTP_IGNORE_REQUEST_TOTAL_LENGTH 0
 #endif
 
+/* Missing in MinGW */
+#ifndef WINHTTP_FLAG_SECURE_PROTOCOL_TLS1
+#define WINHTTP_FLAG_SECURE_PROTOCOL_TLS1 0x00000080
+#endif
+
+#ifndef WINHTTP_FLAG_SECURE_PROTOCOL_TLS1_1
+#define WINHTTP_FLAG_SECURE_PROTOCOL_TLS1_1 0x00000200
+#endif
+
+#ifndef WINHTTP_FLAG_SECURE_PROTOCOL_TLS1_2
+#define WINHTTP_FLAG_SECURE_PROTOCOL_TLS1_2 0x00000800
+#endif
```

Comment:
Hm, some similar constants are defined [over here](https://github.com/libgit2/libgit2/blob/89c332e41b12a72d89de40d63bc568c56a2c336a/deps/winhttp/winhttp.h#L442). Perhaps it makes sense to move these there as well?

Reply:
Good eye, thanks @matklad, I had forgotten about that file entirely.

Reply:
😢 In fact, those are for mingw/mingw32, which needs the whole lib infrastructure that's in `deps/winhttp`.

mingw64 does not use that `deps/winhttp` stuff, it has its own headers for winhttp...  however, they're outdated.  so we actually need to include the constants for mingw64 as well.  (sigh.)

Reply:
>so we actually need to include the constants for mingw64 as well. (sigh.)

Hm, I wonder if the headers for msvc on old windows versions (XP and win7) could also be "outdated"? So perhaps the original code with `ifndef/define` was actually correct? :-)

Reply:
Yep.  Bringing those back.